src/kOS.Safe/Screen/ScreenBufferLine.cs

```diff
@@ -67,13 +68,22 @@ public ScreenBufferLine(int size)
         /// </summary>
         public void ArrayCopyFrom(char[] source, int sourceStart, int destinationStart, int length = -1)
         {
-            Array.Copy(source, sourceStart, charArray, destinationStart, (length >= 0 ? length : source.Length));
+            int maxSafeLength = (charArray.Length - destinationStart);
+
+            int effectiveLength = (length < 0 ? source.Length : length);
+            if (effectiveLength > maxSafeLength)
+            {
+                effectiveLength = maxSafeLength;
+            }
+
+            Array.Copy(source, sourceStart, charArray, destinationStart, effectiveLength);
             TouchTime();
         }
 
         /// <summary>
         /// Perform the Array.Copy() function.
-        /// Copies a subrange of values from a source array into this array.
+        /// Copies a subrange of values from a source array into this array,
+        /// with safety checks to truncate the copy length if it's out of range.
```

Comment:
you added this to the summary without actually changing any of the code in the method like you did above? is that correct?